common/extensions/api/brave_playlists.json

```diff
@@ -0,0 +1,222 @@
+/* Copyright (c) 2019 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+[
+  {
+    "namespace": "bravePlaylists",
+    "description": "Use the <code>chrome.bravePlaylists</code> API to get/set playlist",
+    "compiler_options": {
+      "implemented_in": "brave/browser/extensions/api/brave_playlists_api.h"
+    },
+    "events": [
+      {
+        "name": "onPlaylistsChanged",
+        "type": "function",
+        "description": "Fired when playlists is changed",
+        "parameters": [
+          {
+            "name": "changeType",
+            "type": "string",
+            "description": "Playlists change status(add, delete, abort, ready)"
+          },
+          {
+            "name": "id",
+            "type": "string",
+            "description": "playlist id of added or deleted playlist"
+          }
+        ]
+      },
+      {
+        "name": "onInitialized",
+        "type": "function",
+        "description": "Fired when playlists backend is initialized",
+        "parameters": []
+      }
+    ],
+    "types": [
+      {
+        "id": "PlaylistInfo",
+        "type": "object",
+        "properties": {
+          "id": {
+            "type": "string",
+            "description": "id for each playlist"
+          },
+          "playlistName": {
+            "type": "string",
+            "description": "Name for playlist"
+          },
+          "thumbnailPath": {
+            "type": "string",
+            "description": "thumbnail path for playlist. if this is empty, thumbnail is not fetched yet."
```

Comment:
We can distinguish them by status from `onPlaylistsChanged`.
When `added` type is received, it means playlists is added but thumbnail and songs are not fetched.
`thumbnail_ready` means, thumbnail is fetched.
`play_ready` means, songs are fetched.

I'm trying to add one more api `recoverPlaylst(id)`.
It will try to check current playlist data and will fetch thumbnail or generate media file when it doesn't have them.
Extension could call this new api if it received thumbnail failed/aborted message.